src/net.h

```diff
@@ -955,7 +956,7 @@ class CConnman
     bool AlreadyConnectedToAddress(const CAddress& addr);
 
     bool AttemptToEvictConnection();
-    CNode* ConnectNode(CAddress addrConnect, const char *pszDest, bool fCountFailure, ConnectionType conn_type);
+    CNode* ConnectNode(CAddress addrConnect, const char *pszDest, bool fCountFailure, ConnectionType conn_type) EXCLUSIVE_LOCKS_REQUIRED(!m_i2p_sessions_mutex);
```

Comment:
> would adding this lock annotation only to the `ConnectNode` declaration suffice?

No, all callers of `EXCLUSIVE_LOCKS_REQUIRED(!m)` must also be tagged with the same annotation. Otherwise:

```
net.cpp:1995:20: error: calling function 'ConnectNode' requires negative capability '!m_unused_i2p_sessions_mutex' [-Werror,-Wthread-safety-analysis]
    CNode* pnode = ConnectNode(addrConnect, pszDest, fCountFailure, conn_type);
                   ^
```

Added `AssertLockNotHeld()` calls to methods marked with `EXCLUSIVE_LOCKS_REQUIRED(!m_unused_i2p_sessions_mutex)`